pkg/certificate/peer.go

```diff
@@ -43,23 +45,29 @@ type Peer struct {
 	Log      *zap.Logger
 	Identity *identity.FullIdentity
 
-	Server *server.Server
+	Server          *server.Server
+	AuthorizationDB *authorization.DB
 
 	// services and endpoints
-	Certificates struct {
-		AuthorizationDB *authorization.DB
-		Endpoint        *Endpoint
+	Certificate struct {
+		Endpoint *Endpoint
+	}
+
+	Authorization struct {
+		Listener net.Listener
+		Service  *authorization.Service
+		Endpoint *authorization.Endpoint
 	}
 }
 
 // New creates a new certificates peer.
-func New(log *zap.Logger, ident *identity.FullIdentity, ca *identity.FullCertificateAuthority, authorizationDB *authorization.DB, revocationDB *revocation.DB, config *Config) (*Peer, error) {
+func New(log *zap.Logger, ident *identity.FullIdentity, ca *identity.FullCertificateAuthority, authorizationDB *authorization.DB, revocationDB *revocation.DB, config *Config) (_ *Peer, err error) {
```

Comment:
I think that there is no need to declare `err` in the return values.

Reply:
`err` is required for line 90:
```
	peer.Authorization.Listener, err = net.Listen("tcp", config.AuthorizationAddr)
```

Reply:
For those cases, the var should be declared in the function body.
Named return parameters should be only used for at least one of these cases

* documentation purpose.
* Needed because the value must be referenced in a `defer` statement.

They shouldn't be used only for the sake of declaring a variable.